shared-services/README.md

```diff
@@ -0,0 +1,172 @@
+# DevX 
+
+The `devx` flake provisions a development environment suitable for working on a `haskell.nix` project.
+
+It includes a complete haskell toolchain as well as all the native dependencies needed to work with other haskell IOG repositories.
+
+# Test that `devx` can build your Haskell project 
+
+Pick your `ghc` version (one of `ghc8107`, `ghc901`, `ghc924`), and enter the shell:
+```
+nix develop github:zeme-iohk/devx#ghc8107 --no-write-lock-file
+```
+You may now build and test your project:
+```
+cabal update
+cabal build all -j 
+cabal test all 
+```
+
+If you wish to use `devx` in your project:
+- If you have a new repository then [click here](#integration-with-a-new-repository) for instructions.
+- If you wish to migrate your existing nix code then [click here](#integration-with-an-existing-repository) instead.
+
+# Integration with a new repository 
+
+If this is a new repository, you should use the flake template.
+
+```
+nix flake init --template github:zeme-iohk/devx
+```
+
+Two files will be created in the top-level: `flake.nix` and `devx.nix`.
```

Comment:
Why a separate file instead of just a single top-level function in flake.nix that can be modified? Or we could use the module system/flake-parts and then the user can set whatever `imports` they want (perhaps auto-discovered from the filesystem?)